torchvision/transforms/v2/functional/_utils.py

```diff
@@ -1,8 +1,136 @@
+        raise TypeError(
+            f"Dispatcher '{dispatcher.__name__}' already has a kernel registered for type '{datapoint_cls.__name__}'."
+        )
+
+    def decorator(kernel):
+        registry[datapoint_cls] = _kernel_datapoint_wrapper(kernel) if datapoint_wrapper else kernel
+        return kernel
+
+    return decorator
+
+
+def register_kernel(dispatcher, datapoint_cls):
+    return _register_kernel_internal(dispatcher, datapoint_cls, datapoint_wrapper=False)
+
+
+def _get_kernel(dispatcher, datapoint_cls):
+    registry = _KERNEL_REGISTRY.get(dispatcher)
+    if not registry:
+        raise ValueError(f"No kernel registered for dispatcher '{dispatcher.__name__}'.")
+
+    if datapoint_cls in registry:
+        return registry[datapoint_cls]
+
+    for registered_cls, kernel in registry.items():
+        if issubclass(datapoint_cls, registered_cls):
+            return kernel
+
+    return _noop
+
+
+# Everything below this block is stuff that we need right now, since it looks like we need to release in an intermediate
+# stage. See https://github.com/pytorch/vision/pull/7747#issuecomment-1661698450 for details.
+
+
+# In the future, the default behavior will be to error on unsupported types in dispatchers. The noop behavior that we
+# need for transforms will be handled by _get_kernel rather than actually registering no-ops on the dispatcher.
+# Finally, the use case of preventing users from registering kernels for our builtin types will be handled inside
+# register_kernel.
+def _register_explicit_noop(*datapoints_classes, warn_passthrough=False):
+    """
+    Although this looks redundant with the no-op behavior of _get_kernel, this explicit registration prevents users
+    from registering kernels for builtin datapoints on builtin dispatchers that rely on the no-op behavior.
+
+    For example, without explicit no-op registration the following would be valid user code:
+
+    .. code::
+        from torchvision.transforms.v2 import functional as F
+
+        @F.register_kernel(F.adjust_brightness, datapoints.BoundingBox)
+        def lol(...):
+            ...
+    """
+
+    def decorator(dispatcher):
+        for cls in datapoints_classes:
+            msg = (
+                f"F.{dispatcher.__name__} is currently passing through inputs of type datapoints.{cls.__name__}. "
+                f"This will likely change in the future."
+            )
+            register_kernel(dispatcher, cls)(functools.partial(_noop, __msg__=msg if warn_passthrough else None))
+        return dispatcher
+
+    return decorator
+
+
+def _noop(inpt, *args, __msg__=None, **kwargs):
+    if __msg__:
+        warnings.warn(__msg__, UserWarning, stacklevel=2)
+    return inpt
+
+
+# TODO: we only need this, since our default behavior in case no kernel is found is passthrough. When we change that
+# to error later, this decorator can be removed, since the error will be raised by _get_kernel
+def _register_unsupported_type(*datapoints_classes):
```

Comment:
I was a bit surprised to see this so to clarify (for myself): this is only used on `normalize` (which was already erroring on PIL images) or on the "shape getter" which are fake kernels which can't pass-through anyway (they don't return a transformed input, they return shapes).
Can you confirm this is correct @pmeier ?

Reply:
Can confirm.